### PR TITLE
Multi-partition spec support fix in Avro to ORC

### PR DESCRIPTION
**Pull request by abti**
Fix in partition spec parsing, DDL and DML generation for multi-partition spec support

### CODE REVIEW

**Comment by abti** [comment]
Minor fixes, self approving.

File: gobblin-data-management/src/main/java/gobblin/data/management/conversion/hive/writer/HiveQueryExecutionWriter.java [hunk 1]
```diff
@@ -45,11 +45,9 @@ public void write(QueryBasedHiveConversionEntity hiveConversionEntity) throws IO
       this.hiveJdbcConnector.executeStatements(conversionQueries.toArray(new String[conversionQueries.size()]));
       EventWorkunitUtils.setEndConversionDDLExecuteTimeMetadata(this.workUnit, System.currentTimeMillis());
     } catch (SQLException e) {
-      if (null != conversionQueries) {
-        log.warn("Dumping queries attempted by the writer.");
-        for (String conversionQuery : conversionQueries) {
-          log.warn("Conversion query attempted by Hive Query writer: " + conversionQuery);
-        }
+      log.warn("Dumping queries attempted by the writer.");
```
Comment (pcadabam) [inline gobblin-data-management/src/main/java/gobblin/data/management/conversion/hive/writer/HiveQueryExecutionWriter.java]: Failed to execute queries?

File: gobblin-data-management/src/main/java/gobblin/data/management/conversion/hive/query/HiveAvroORCQueryGenerator.java [hunk 2]
```diff
@@ -756,7 +756,13 @@ public static String generateTableMappingDML(Schema inputAvroSchema,
     if (optionalPartitionDMLInfo.isPresent()) {
       if (optionalPartitionDMLInfo.get().size() > 0) {
         dmlQuery.append("WHERE ");
+        boolean isFirstPartitionSpec = true;
         for (Map.Entry<String, String> partition : optionalPartitionDMLInfo.get().entrySet()) {
+          if (isFirstPartitionSpec) {
+            isFirstPartitionSpec = false;
+          } else {
+            dmlQuery.append(" and ");
```
Comment (pcadabam) [inline gobblin-data-management/src/main/java/gobblin/data/management/conversion/hive/query/HiveAvroORCQueryGenerator.java]: `AND` we use upper case everywhere else

File: gobblin-data-management/src/main/java/gobblin/data/management/conversion/hive/writer/HiveQueryExecutionWriter.java [hunk 3]
```diff
@@ -35,12 +38,17 @@
...
   public void write(QueryBasedHiveConversionEntity hiveConversionEntity) throws IOException {
+    List<String> conversionQueries = null;
     try {
-      List<String> conversionQueries = hiveConversionEntity.getQueries();
+      conversionQueries = hiveConversionEntity.getQueries();
       EventWorkunitUtils.setBeginConversionDDLExecuteTimeMetadata(this.workUnit, System.currentTimeMillis());
       this.hiveJdbcConnector.executeStatements(conversionQueries.toArray(new String[conversionQueries.size()]));
       EventWorkunitUtils.setEndConversionDDLExecuteTimeMetadata(this.workUnit, System.currentTimeMillis());
     } catch (SQLException e) {
+      log.warn("Dumping queries attempted by the writer.");
```
Comment (pcadabam) [inline gobblin-data-management/src/main/java/gobblin/data/management/conversion/hive/writer/HiveQueryExecutionWriter.java]: `Failed executing queries` is more descriptive

Comment (abti) [inline gobblin-data-management/src/main/java/gobblin/data/management/conversion/hive/writer/HiveQueryExecutionWriter.java]: Thanks for the review. Made the changes.

**Pull request merged**